mackerel-plugin-mssql/lib/mssql_windows.go

```diff
@@ -0,0 +1,183 @@
+	stat["stats_sql_recompilations"] = float64(sqlStatistics[0].SQLReCompilationsPersec)
+	stat["stats_connections"] = float64(generalStatistics[0].UserConnections)
+	stat["stats_lock_waits"] = float64(generalStatistics[0].SQLTraceIOProviderLockWaits)
+	stat["stats_procs_blocked"] = float64(generalStatistics[0].Processesblocked)
+	stat["access_page_splits"] = float64(accessMethods[0].PageSplitsPersec)
+	return stat, nil
+}
+
+// GraphDefinition interface for mackerelplugin
+func (m MSSQLPlugin) GraphDefinition() map[string](mp.Graphs) {
+	labelPrefix := strings.Title(strings.Replace(m.MetricKeyPrefix(), "mssql", "MSSQL", -1))
+	return map[string](mp.Graphs){
+		"buffer": mp.Graphs{
+			Label: labelPrefix + " Buffer",
+			Unit:  mp.UnitInteger,
+			Metrics: []mp.Metrics{
+				{
+					Name:    "buffer_cache_hit_ratio",
+					Label:   "Cache Hit Ratio",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "buffer_page_life_expectancy",
+					Label:   "Page Life Expectancy",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "buffer_checkpoint_pages",
+					Label:   "Checkpoint Pages",
+					Diff:    false,
+					Stacked: true,
+				},
+			},
+		},
+		"stats": mp.Graphs{
+			Label: labelPrefix + " Stats",
+			Unit:  mp.UnitInteger,
+			Metrics: []mp.Metrics{
+				{
+					Name:    "stats_batch_requests",
+					Label:   "Batch Requests",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "stats_sql_compilations",
+					Label:   "SQL Compilations",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "stats_sql_recompilations",
+					Label:   "SQL Re-Compilations",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "stats_connections",
+					Label:   "Connections",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "stats_lock_waits",
+					Label:   "Lock Waits",
+					Diff:    false,
+					Stacked: true,
+				},
+				{
+					Name:    "stats_procs_blocked",
+					Label:   "Procs Blocked",
+					Diff:    false,
+					Stacked: true,
+				},
+			},
+		},
+		"access": mp.Graphs{
+			Label: labelPrefix + " Access",
+			Unit:  mp.UnitInteger,
+			Metrics: []mp.Metrics{
+				{
+					Name:    "access_page_splits",
+					Label:   "Page Splits",
+					Diff:    false,
+					Stacked: true,
+				},
+			},
+		},
+	}
+}
+
+// Do the plugin
+func Do() {
+	optPrefix := flag.String("metric-key-prefix", "mssql", "Metric key prefix")
+	optInstance := flag.String("instance", "SQLSERVER", "Instance name of MSSQL(SQLSERVER or SQLEXPRESS)")
+	optTempfile := flag.String("tempfile", "", "Temp file name")
+	flag.Parse()
+
+	var plugin MSSQLPlugin
+	plugin.prefix = *optPrefix
+	plugin.instance = strings.ToUpper(*optInstance)
+	helper := mp.NewMackerelPlugin(plugin)
+
+	if *optTempfile != "" {
+		helper.Tempfile = *optTempfile
+	} else {
+		helper.Tempfile = "/tmp/mackerel-plugin-mssql"
```

Comment:
Now default Tempfile path is set by helper automatically, so we don't have to set this default value.
ref: https://github.com/mackerelio/go-mackerel-plugin-helper/pull/22 (This is go-mackerel-plugin*-helper* 's p-r, but this feature is also backported to go-mackerel-plugin.

====

(But this is unnecessary comment if `--tempfile` option itself is removed)